packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/DataSourceInlineEditIdentityJoins.tsx

```diff
@@ -19,12 +20,14 @@ export const DataSourceInlineEditIdentityJoins: FC<DataSourceInlineEditIdentityJ
   dataSource,
   onSave,
   onCancel,
+  canEdit = true,
 }) => {
+  const { project } = useDefinitions();
   const [uiMode, setUiMode] = useState<"view" | "edit" | "add">("view");
   const [editingIndex, setEditingIndex] = useState<number>(-1);
 
   const permissions = usePermissions();
-  const canEdit = permissions.editDatasourceSettings;
+  canEdit = canEdit && permissions.check("editDatasourceSettings", project);
```

Comment:
This should use the datasource's projects, not the project set in definitions

Reply:
same update as above